modules/geo-layers/src/tile-layer/tile-2d-traversal.js

```diff
@@ -0,0 +1,132 @@
+import {CullingVolume, Plane, AxisAlignedBoundingBox} from '@math.gl/culling';
+
+const TILE_SIZE = 512;
+// number of world copies to check
+const MAX_MAPS = 3;
+
+class OSMNode {
+  constructor(x, y, z) {
+    this.x = x;
+    this.y = y;
+    this.z = z;
+  }
+
+  get children() {
+    if (!this._children) {
+      const x = this.x * 2;
+      const y = this.y * 2;
+      const z = this.z + 1;
+      this._children = [
+        new OSMNode(x, y, z),
+        new OSMNode(x, y + 1, z),
+        new OSMNode(x + 1, y, z),
+        new OSMNode(x + 1, y + 1, z)
+      ];
+    }
+    return this._children;
+  }
+
+  update(params) {
+    const {viewport, cullingVolume, elevationBounds, minZ, maxZ, offset} = params;
+    const boundingVolume = this.getBoundingVolume(elevationBounds, offset);
+
+    // First, check if this tile is visible
+    const isInside = cullingVolume.computeVisibility(boundingVolume);
+    if (isInside < 0) {
+      return false;
+    }
+
+    // Avoid loading overlapping tiles - if a descendant is requested, do not request the ancester
+    if (!this.childVisible) {
+      let {z} = this;
+      if (z < maxZ && z >= minZ) {
+        // Adjust LOD by distance to camera
+        const distance =
+          (boundingVolume.distanceTo(viewport.cameraPosition) * viewport.scale) / viewport.height;
+        z += Math.floor(Math.log2(distance));
```

Comment:
why do you need apply `* viewport.scale / viewport.height` to  `(boundingVolume.distanceTo(viewport.cameraPosition)`? Maybe add a comment about your calculation?